src/main/java/org/jabref/gui/DuplicateResolverDialog.java

```diff
@@ -39,108 +33,91 @@
+                both = new ButtonType(Localization.lang("Import and keep old entry"), ButtonData.APPLY);
+                me = new MergeEntries(one, two, Localization.lang("Old entry"),
+                                      Localization.lang("From import"), frame.getCurrentBasePanel().getBibDatabaseContext().getMode());
+                break;
         }
-        options.add(first);
-        options.add(second);
-        options.add(both);
-        options.add(merge);
-        options.add(Box.createHorizontalStrut(5));
-        options.add(cancel);
-        options.add(helpButton);
-
-        first.addActionListener(e -> buttonPressed(DuplicateResolverResult.KEEP_LEFT));
-        second.addActionListener(e -> buttonPressed(DuplicateResolverResult.KEEP_RIGHT));
-        both.addActionListener(e -> buttonPressed(DuplicateResolverResult.KEEP_BOTH));
-        merge.addActionListener(e -> buttonPressed(DuplicateResolverResult.KEEP_MERGE));
-        if (removeExact != null) {
-            removeExact.addActionListener(e -> buttonPressed(DuplicateResolverResult.AUTOREMOVE_EXACT));
+        if (removeExactVisible) {
+            this.getDialogPane().getButtonTypes().add(removeExact);
         }
 
-        cancel.addActionListener(e -> buttonPressed(DuplicateResolverResult.BREAK));
-        addWindowListener(new WindowAdapter() {
-            @Override
-            public void windowClosing(WindowEvent e) {
-                buttonPressed(DuplicateResolverResult.BREAK);
-            }
-        });
-
-        getContentPane().add(CustomJFXPanel.wrap(new Scene(me)));
-        getContentPane().add(options, BorderLayout.SOUTH);
-        pack();
-
-        WindowLocation pw = new WindowLocation(this, JabRefPreferences.DUPLICATES_POS_X,
-                JabRefPreferences.DUPLICATES_POS_Y, JabRefPreferences.DUPLICATES_SIZE_X,
-                JabRefPreferences.DUPLICATES_SIZE_Y);
-        pw.displayWindowAtStoredLocation();
+        this.getDialogPane().getButtonTypes().addAll(first, second, both, merge, cancel, helpButton);
 
-        both.requestFocus();
-    }
+        BorderPane borderPane = new BorderPane(me);
+        borderPane.setBottom(options);
 
-    private void buttonPressed(DuplicateResolverResult result) {
-        status = result;
-        dispose();
-    }
+        this.setResultConverter(button -> {
+            if (button.equals(first)) {
+                return DuplicateResolverResult.KEEP_LEFT;
+            }
+            if (button.equals(second)) {
+                return DuplicateResolverResult.KEEP_RIGHT;
+            }
+            if (button.equals(both)) {
+                return DuplicateResolverResult.KEEP_BOTH;
+            }
+            if (button.equals(merge)) {
+                return DuplicateResolverResult.KEEP_MERGE;
+            }
+            if (button.equals(removeExact)) {
+                return DuplicateResolverResult.AUTOREMOVE_EXACT;
+            }
+            return DuplicateResolverResult.BREAK;
```

Comment:
The standard convention in JavaFX is to return `null` if the user aborted the process. I think, we should adhere to this convention instead of introducing the `BREAK` return value.